engine/immediate_task_conrrent_processor.go

```diff
@@ -426,3 +425,29 @@ func (w *immediateTaskConcurrentProcessor) composeHttpError(
 
 	return statusCode, details, fmt.Errorf("statusCode: %v, errMsg: %w, responseBody: %v", statusCode, err, responseBody)
 }
+
+func (w *immediateTaskConcurrentProcessor) processLocalQueueMessagesTask(
+	ctx context.Context, task persistence.ImmediateTask,
+) error {
+	resp, err := w.store.ProcessLocalQueueMessages(ctx, persistence.ProcessLocalQueueMessagesRequest{
+		TaskShardId:  task.ShardId,
+		TaskSequence: task.GetTaskSequence(),
+
+		ProcessExecutionId: task.ProcessExecutionId,
+
+		Messages: task.ImmediateTaskInfo.LocalQueueMessageInfo,
+	})
+	if err != nil {
+		return err
+	}
+
+	// TODO: need a better way to define the NotifyNewImmediateTasks method
```

Comment:
I don't quite follow, can you expand more on that?

Reply:
This comment has been removed...

Reply:
Oh I found it still exists, maybe there were two places with this comment and I only deleted one. Deleted it now.